File: bindings/python/python/maxt/examples/candles_history.py
"""Read public Binance Spot candles and prepare a paged history request."""

from __future__ import annotations

import asyncio

from maxt import (
    BinanceAdapter,
    CandleRequest,
    Client,
    Exchange,
    HistoryRequest,
    Interval,
    Market,
)


async def main() -> None:
    market = Market.spot(Exchange.BINANCE, "BTC", "USDT")
    client = Client(BinanceAdapter.spot())
    candles = await client.candles(CandleRequest(market, Interval.MIN1, limit=5))
    for candle in candles:
        print(f"{candle.open_time}: close={candle.close} volume={candle.volume}")

    # Feed `page.next` into `cursor` only when a previous private response has
    # one. The request itself performs no account or network action.
    history = HistoryRequest(market, limit=100)
    print(f"private history request prepared: {history}")


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/dart/example/candles_history.dart
import 'package:maxt/maxt.dart';

Future<void> main() async {
  await Maxt.initialize();
  try {
    final market = Market.spot(Exchange.binance, 'BTC', 'USDT');
    final client = Client(BinanceAdapter.spot());
    final candles = await client.candles(
      CandleRequest(market, Interval.min1, limit: 5),
    );
    for (final candle in candles) {
      print(
        '${candle.openTime}: close=${candle.close} volume=${candle.volume}',
      );
    }

    // Reuse a previous page cursor only when the previous response has one.
    final history = HistoryRequest(market, limit: 100);
    print('Private history request prepared for ${history.market}');
  } finally {
    await Maxt.dispose();
  }
}
